core/cmd/client.go

```diff
@@ -386,6 +390,39 @@ func (n ChainlinkRunner) Run(ctx context.Context, app chainlink.Application) err
 	return errors.WithStack(g.Wait())
 }
 
+func sentryInit(cfg config.BasicConfig) error {
+	sentrydsn := cfg.SentryDSN()
+	if sentrydsn == "" {
+		// Do not initialize sentry at all if the DSN is missing
+		return nil
+	}
+
+	var sentryenv string
+	if env := cfg.SentryEnvironment(); env != "" {
+		sentryenv = env
+	} else if cfg.Dev() {
+		sentryenv = "dev"
+	} else {
+		sentryenv = "prod"
+	}
+
+	var sentryrelease string
+	if release := cfg.SentryRelease(); release != "" {
+		sentryrelease = release
+	} else {
+		sentryrelease = static.Version
+	}
+
+	return sentry.Init(sentry.ClientOptions{
```

Comment:
In theory yes, but in implementation no, because we use an `EmptyRunner` in tests. Also, empirically, the gin stuff would have the same problem, but has been here over a year without flagging any races.